server/src/com/thoughtworks/go/server/materials/postcommit/github/GitHubRepository.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2017 ThoughtWorks, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.thoughtworks.go.server.materials.postcommit.github;
+
+import java.util.Objects;
+
+public class GitHubRepository {
+    String url;
+    String fullName;
+
+    public String getUrl() {
+        return url;
+    }
+
+    public void setUrl(String url) {
```

Comment:
From what I can see setters are only used within tests. The team prefers using constructor injection instead of setters, and making the instance variables final —

```
class GitHubRepository {
  private final String url;
  private final String fullName;
  public final GitHubRepository() {
    // no-op constructor for GSON
  }

  public final GitHubRepository(String url, String fullName) {
    this.url = url;
    this.fullName = fullName;
  }
}
```

Ditto for `GitHubPushEvent` above.